File: flatseeker-backend/src/main/java/com/yakzhanov/flatseeker/model/dict/ProcessStatus.java
package com.yakzhanov.flatseeker.model.dict;

import javax.persistence.Entity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
public class ProcessStatus extends DictParent {

    private String color;
    private boolean active;

}

File: flatseeker-ui/src/app/model/dict/ProcessStatus.ts
export class ProcessStatus {
  public key: string;
  public title: string;
  public color: string;
  public active: boolean;

  constructor(key: string, title: string, color: string, active: boolean) {
    this.key = key;
    this.title = title;
    this.color = color;
    this.active = active;
  }

  public static mock() {
    return new ProcessStatus("", "", "", false);
  }
}
